server/src/main/java/org/opensearch/cluster/metadata/MetadataIndexUpgradeService.java

```diff
@@ -134,20 +135,35 @@ boolean isUpgraded(IndexMetadata indexMetadata) {
     private void checkSupportedVersion(IndexMetadata indexMetadata, Version minimumIndexCompatibilityVersion) {
         if (indexMetadata.getState() == IndexMetadata.State.OPEN
             && isSupportedVersion(indexMetadata, minimumIndexCompatibilityVersion) == false) {
-            throw new IllegalStateException(
-                "The index ["
-                    + indexMetadata.getIndex()
-                    + "] was created with version ["
-                    + indexMetadata.getCreationVersion()
-                    + "] but the minimum compatible version is ["
-
-                    + minimumIndexCompatibilityVersion
-                    + "]. It should be re-indexed in OpenSearch "
-                    + minimumIndexCompatibilityVersion.major
-                    + ".x before upgrading to "
-                    + Version.CURRENT
-                    + "."
-            );
+            if (minimumIndexCompatibilityVersion.equals(LegacyESVersion.V_7_0_0)) {
+                // This branch can be removed when LegacyESVersion is removed
```

Comment:
Can we create an issue and add to the comment?

Reply:
I don't think we will every actually remove this code...it will just stay orphaned on the 2.x branch when the world moves ahead. But in any case, there's an issue for removing this constant from 3.0 (#2747) so the compiler would complain if this code needed to be updated.